### PR TITLE
feat: show Red Hat SSO status icon with tooltip in status bar

### PR DESCRIPTION
**Pull request by dgolovin**
Fix: #78

It there are not sessions saved in secrets store then after start the status bar item has state:
1. tooltip test is `Red Hat SSO: Logged Out`
2. command executed on click starts login workflow

If there are sessions in secrets store and they are loaded during initialization then bar item has:
1. Tooltip `Red Hat SSO: Logged in as email@address.com`
2. Command is not set

Status bar item also gets updated accordingly if log in/log out status changes during the podman-desktop session.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -203,13 +204,22 @@ async function removeSession(sessionId: string): Promise<void> {
...
+  statusBarItem.show()
+  context.subscriptions.push(statusBarItem);
   if (!authenticationServicePromise) {
     authenticationServicePromise = RedHatAuthenticationService.build(context, getAuthConfig())
       .then(service => {
         context.subscriptions.push(service);
-        service.initialize();
-        return service;
-    });
+        return service.initialize().then(() => service);
```
Comment (benoitf) [inline src/extension.ts:214]: I think here we reached from readability too many then then then

Comment (dgolovin) [reply src/extension.ts:214]: I'll move it out to async function.

**Comment by vrothberg** [comment]
Thank you!  At the moment, I see some issues:
1. The icon is clickable but there is no action. It could open the Authentication page.
2. Hovering over the icon, it takes a surprisingly long time for the pop-up to appear. Maybe we can reduce the time for it to show up?
3. The icon does not change once signed in.

Screenshot 2024-04-02 at 09 42 01

**Comment by dgolovin** [comment]
When logged out, It is clickable and starts login workflow for Red Hat SSO.
When logged in, it is clickable (because it is always clickable in status bar), but no command is assigned.
I cannot navigate to Authentication Settings right now because `navigation' API does not support navigation to that page.
Adding `navigation.navigateToAuthenticationSettings` to Podman Desktop API is straightforward if we want do that (cc: @benoitf @deboer-tim).

Hint delay is OS dependent and implementation is on podman desktop side.

I can have two icons one for logged in and one for logged out, but I what kind of change you have in mind?

**Comment by benoitf** [comment]
> Adding navigation.navigateToAuthenticationSettings` to Podman Desktop API is straightforward if we want do that

you need to open a PR ASAP as we're about to cut v1.9.0

**Comment by dgolovin** [comment]
PR for navigation to authentication settings is merged.

**Comment by vrothberg** [comment]
Clicking on the icon in the status does not yet trigger an action for me.

**Comment by dgolovin** [comment]
I could replicate it at some point, but after rebuilding everything it works for me.

What I noticed, there is a slight delay between click and opening page in browser.

**Comment by dgolovin** [comment]
> Clicking on the icon in the status does not yet trigger an action for me.
@vrothberg found the case when I can replicate it. It happens when you exit podman desktop while you logged into RedHat SSO, then you start after after a moment in time when refresh token is expired and you are not logged in anymore, but your podman vm and registry access is configured. In this case extension does not trigger login sequence, because it is not required :)

**Comment by dgolovin** [comment]
> Clicking on the icon in the status does not yet trigger an action for me.
@vrothberg should work with latest commit but you will see this issue https://github.com/containers/podman-desktop/issues/6664 when trigger login form status bar.

**Review by vrothberg — COMMENTED** [review]
Thanks!  Tested successfully

**Pull request merged**